### PR TITLE
[url_launcher] Migrate to new pubspec plugins schema and endorse url_launcher_web

### PR DESCRIPTION
**Pull request by amirh**
Migrates url_launcher to use the new pubspec plugins schema, and endorses url_launcher_web as the default web implementation.

https://github.com/flutter/flutter/issues/46253

### CODE REVIEW

File: packages/url_launcher/url_launcher/pubspec.yaml [hunk 1]
```diff
@@ -7,9 +7,12 @@ version: 5.2.7
...
-    pluginClass: UrlLauncherPlugin
+    platforms:
+      android:
+        package: io.flutter.plugins.urllauncher
+        pluginClass: UrlLauncherPlugin
+      ios:
+        pluginClass: FLTUrlLauncherPlugin
 
 dependencies:
   flutter:
```
Comment (ditman) [inline packages/url_launcher/url_launcher/pubspec.yaml]: No new dependency on `url_launcher_web` yet?

Comment (amirh) [reply packages/url_launcher/url_launcher/pubspec.yaml]: Good point, I guess we might as well do it in this PR as it is already gated on a new enough Flutter

File: packages/url_launcher/url_launcher/pubspec.yaml [hunk 2]
```diff
@@ -3,18 +3,24 @@ description: Flutter plugin for launching a URL on Android and iOS. Supports
...
   plugin:
-    androidPackage: io.flutter.plugins.urllauncher
-    iosPrefix: FLT
-    pluginClass: UrlLauncherPlugin
+    platforms:
+      android:
+        package: io.flutter.plugins.urllauncher
+        pluginClass: UrlLauncherPlugin
+      ios:
+        pluginClass: FLTUrlLauncherPlugin
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher/pubspec.yaml]: Nit: FLTURLLauncherPlugin (same for the filename). The old capitalization was wrong for ObjC, but there was no way to have it be different for ObjC and Java. Now that there is, it should be fixed.

File: packages/url_launcher/url_launcher/pubspec.yaml [hunk 3]
```diff
@@ -13,11 +13,14 @@ flutter:
...
       ios:
         pluginClass: FLTUrlLauncherPlugin
+      web:
+        default_package: url_launcher_web
 
 dependencies:
   flutter:
     sdk: flutter
   url_launcher_platform_interface: ^1.0.1
+  url_launcher_web: any
```
Comment (ditman) [inline packages/url_launcher/url_launcher/pubspec.yaml]: I'd be specific on this one, but we'll see how 'any' goes for a while!

Comment (amirh) [reply packages/url_launcher/url_launcher/pubspec.yaml]: See discussion on https://flutter.dev/go/federated-plugins we want this edge to be any

File: packages/url_launcher/url_launcher/pubspec.yaml (hunk 1 above)
Comment (amirh) [reply packages/url_launcher/url_launcher/pubspec.yaml]: done

**Pull request merged**